### PR TITLE
Doc update v4

### PR DESCRIPTION
**Pull request by isaacsas**
@gzagatti I made (am making) some more updates. I can't push to your branch either as I think you have this disabled on your fork.

Let's just work in this version. If you can't push to it let me know and I'll try to get that straigtened out.

### CODE REVIEW

File: docs/src/tutorials/point_process_simulation.md [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+distinct from our state variable vector, `u`. When `poisson_process` fires
+`u[1]` is altered, and as the rate for `seasonal_process` depends on it we have
+that the dependencies of `poisson_process` are `[1,2]`. In contrast, the rate of
+`poisson_process` is independent of `u[2]` which `seasonal_process` modifies,
+and hence the dependencies of `seasonal_process` are only `[2]`.
+
+Therefore, we obtain the following dependency graph:
+
+```@example tpp-tutorial
+dep_graph = [[1,2], [2]]
```
Comment (isaacsas) [inline docs/src/tutorials/point_process_simulation.md:234]: @gzagatti I changed this as it seemed reversed in your PR. Can you double check this?

**Comment by isaacsas** [comment]
@gzagatti I am ok to merge this if my updates to the simple TPP tutorial are ok with you. Please see my one question.

I haven't had time to carefully go through your more involve application that adds an interface to PointProcesses, but am ok with merging now that we've moved it to an application and I clarified the language that isn't a basic tutorial

File: docs/src/tutorials/point_process_simulation.md (hunk 1 above)
Comment (gzagatti) [reply docs/src/tutorials/point_process_simulation.md:234]: That was a mistake. Your fix is correct.

**Comment by gzagatti** [comment]
@isaacsas it looks all good to me. I have corrected a few typos and fixed a dangling reference. I can't push to this branch, but you can find my changes [here](https://github.com/gzagatti/JumpProcesses.jl/tree/doc_update_v4). I guess you can just merge. 

I'm good to have the PR merged after these fixes.

**Comment by isaacsas** [comment]
Ok, push the version with your additions as a v5 PR and I can merge it after tests pass tomorrow morning.

**Pull request closed without merging**